### DISCUSSION TITLE
Add ability to pin individual plugin settings

### DISCUSSION THREAD

**Discussion by Syntribos** [runelite/runelite#16784] [Ideas]
I find myself often toggling a few settings in a few different plugins (e.g. always on top in the RuneLite plugin, or various graphics settings in 117HD if I need to save battery life or want performance elsewhere). In larger plugins (like the aforementioned), this can be a bit cumbersome, especially when the setting isn't at the top or bottom of the settings page.

It would be nice to be able to do one of two things. Number 1 is my ideal case, but number 2 would still be a huge QOL improvement if 1 is unfeasible (both are predicated on being able to "pin" settings in individual plugin config):

1. Pinned settings are grouped in the configuration page, either as their own faux "plugin" that you can click into, or a split section at the top with your overall plugin config page, that has a collection of all your pinned settings, broken up by plugin name to keep it clean.
2. Pinned settings appear at the top of each individual plugin config page, allowing you to quickly click into each plugin and change the relevant settings without having to search for 'em

Attaching low effort screenshots for situation 2

![image](https://github.com/runelite/runelite/assets/7717089/a0548e2d-9182-4299-a257-a33ca3aedf67)

![image](https://github.com/runelite/runelite/assets/7717089/1a41d7d5-4e32-4a59-88e7-04c969fc8370)

**Comment by antero111** [comment]
I'd suggest you should check out the Plugin Presets plugin. With that plugin you could make some presets to store e.g. your low performace settings and then switch between the presets and not chance the settings manually on the config tab...

**Reply by Nightfirecat** [reply]
fwiw this is also possible via the config profiles system

**Reply by Syntribos** [reply]
Valid points, and definitely helpful as an interim solution, though I still think this feature would be helpful for quick settings swaps without needing to manage updating settings/installed plugins between multiple profiles when you only want to quickly toggle a setting or two
